### ISSUE TITLE
Python 2->3 Conversion error in GaussianProcessHyperparamOpt

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1814]
The following line is in `GaussianProcessHyperparamOpt`: https://github.com/deepchem/deepchem/blob/master/deepchem/hyper/gaussian_process.py#L189

```
hyper_parameters[hp[0]] = map(int, hyper_parameters[hp[0]])
```

This crashes in Python 3 since `map` no longer returns a list! This needs to be fixed as follows

```
hyper_parameters[hp[0]] = list(map(int, hyper_parameters[hp[0]]))
```

**Comment by rbharath** [comment]
The same bug also appears at https://github.com/deepchem/deepchem/blob/master/deepchem/hyper/gaussian_process.py#L265

**Comment by rbharath** [comment]
I've implemented this change in #1792.

**Comment by nd-02110114** [comment]
I think this issue was fixed #1878

**Comment by rbharath** [comment]
Yep good catch! Going to close
